pkg/bmc/virtualmedia_validation.go

```diff
@@ -0,0 +1,117 @@
+package bmc
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// contains tells whether the array of redfish.VirtualMediaType
+// has the element redfish.VirtualMediaType.
+func contains(array []redfish.VirtualMediaType, element redfish.VirtualMediaType) bool {
+	for _, n := range array {
+		if element == n {
+			return true
+		}
+	}
+	return false
+}
+
+// checkVirtualMedia tells if all virtual media of a given Manager
+// has the correct type and actions to work with virtual media boot.
+func checkVirtualMedia(m *redfish.Manager) error {
+	vmediaList, err := m.VirtualMedia()
+	if err != nil {
+		return err
+
+	}
+	if len(vmediaList) == 0 {
+		return fmt.Errorf("no virtual media found for manager %s", m.ID)
+	}
+
+	var vmediaError error
```

Comment:
The error reporting here is a bit confusing. There are 2 different sources of error messages, but we're looping over the `vmediaList` as well without accumulating those messages. That means the error reported is only going to be for the last item in the slice. Do we expect the slice to always have 0 or 1 items? Would there be any benefit to reporting all of the errors?